Mathlib/Data/ENat/Lattice.lean

```diff
@@ -109,4 +110,121 @@ lemma exists_eq_iSup₂_of_lt_top {ι₁ ι₂ : Type*} {f : ι₁ → ι₂ →
   rw [iSup_prod'] at h ⊢
   exact Prod.exists'.mp (exists_eq_iSup_of_lt_top h)
 
+variable {ι κ : Sort*} {f g : ι → ℕ∞} {s : Set ℕ∞} {a : ℕ∞}
+
+lemma iSup_natCast : ⨆ n : ℕ, (n : ℕ∞) = ⊤ :=
+  (iSup_eq_top _).2 fun _b hb ↦ ENat.exists_nat_gt (lt_top_iff_ne_top.1 hb)
+
+proof_wanted mul_iSup (a : ℕ∞) (f : ι → ℕ∞) : a * ⨆ i, f i = ⨆ i, a * f i
+proof_wanted iSup_mul (f : ι → ℕ∞) (a : ℕ∞) : (⨆ i, f i) * a = ⨆ i, f i * a
+proof_wanted mul_sSup : a * sSup s = ⨆ b ∈ s, a * b
+proof_wanted sSup_mul : sSup s * a = ⨆ b ∈ s, b * a
+
+proof_wanted mul_iInf' (_hinfty : a = ⊤ → ⨅ i, f i = 0 → ∃ i, f i = 0) (_h₀ : a = 0 → Nonempty ι) :
```

Comment:
I'm confused about this statement. `ENat` is a well-order, so `⨅ i, f i = 0 → ∃ i, f i = 0` is unconditionally true, right?

Reply:
@YaelDillies This assumption also occurs elsewhere.